iroh/tests/provide.rs

```diff
@@ -716,3 +727,106 @@ async fn test_custom_request_collection() {
     .expect("timeout")
     .expect("get failed");
 }
+
+#[derive(Clone, Debug)]
+struct CustomAuthHandler;
+
+impl RequestAuthorizationHandler for CustomAuthHandler {
+    fn authorize(
+        &self,
+        _db: Database,
+        token: Option<RequestToken>,
+        _request: &iroh_bytes::protocol::Request,
+    ) -> BoxFuture<'static, anyhow::Result<()>> {
+        async move {
+            match token {
+                Some(token) => {
+                    if token.to_bytes() != vec![1, 2, 3, 4, 5, 6] {
+                        anyhow::bail!("bad token")
+                    }
+                    Ok(())
+                }
+                None => {
+                    anyhow::bail!("give token plz")
+                }
+            }
+        }
+        .boxed()
+    }
+}
+
+#[tokio::test]
+async fn test_token_passthrough() {
+    let rt = test_runtime();
+    let readme = readme_path();
+    let (db, hash) = create_collection(vec![readme.into()]).await.unwrap();
+    let provider = match Node::builder(db)
+        .bind_addr("[::1]:0".parse().unwrap())
+        .custom_auth_handler(CustomAuthHandler)
+        .runtime(&rt)
+        .spawn()
+        .await
+    {
+        Ok(provider) => provider,
+        Err(_) => {
+            // We assume the problem here is IPv6 on this host.  If the problem is
```

Comment:
Just bind to ipv4 please, and make this not just silently fail. This should always work, if not, this is a bad sign